src/Workspaces/Core/Portable/Classification/SyntaxClassification/AbstractEmbeddedLanguageTokenClassifier.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Classification.Classifiers
+{
+    internal abstract class AbstractEmbeddedLanguageTokenClassifier : AbstractSyntaxClassifier
+    {
+        private readonly IEmbeddedLanguageProvider _languageProvider;
+
+        protected AbstractEmbeddedLanguageTokenClassifier(IEmbeddedLanguageProvider languageProvider)
+        {
+            _languageProvider = languageProvider;
+        }
+
+        public sealed override void AddClassifications(Workspace workspace, SyntaxToken token, SemanticModel semanticModel, ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken)
+        {
+            foreach (var language in _languageProvider.GetEmbeddedLanguages())
+            {
+                var classifier = language.Classifier;
+                if (classifier != null)
+                {
+                    var count = result.Count;
+                    classifier.AddClassifications(workspace, token, semanticModel, result, cancellationToken);
```

Comment:
Instead of checking for changed count, would it make sense for `AddClassifications` to return a `bool` instead?

Reply:
Possibly.  though that woudl push the onus on all the implementations to do this work.  As it is 1:1 in sync with the count of the classifications in teh list, i'd prefer to keep the work done only here.